Source/ARTTypes.h

```diff
@@ -72,34 +110,72 @@ typedef NS_ENUM(NSUInteger, ARTRealtimeConnectionEvent) {
     ARTRealtimeConnectionEventClosing,
     ARTRealtimeConnectionEventClosed,
     ARTRealtimeConnectionEventFailed,
+    /**
+     * BEGIN CANONICAL PROCESSED DOCSTRING
+     * An event for changes to connection conditions for which the `ARTRealtimeConnectionState` does not change.
+     * END CANONICAL PROCESSED DOCSTRING
+     */
     ARTRealtimeConnectionEventUpdate
 };
 
 NSString *_Nonnull ARTRealtimeConnectionEventToStr(ARTRealtimeConnectionEvent event);
 
-
-#pragma mark - ARTRealtimeChannelState
-
 /**
- ARTRealtimeChannelState is an enum representing all the Realtime Channel states.
+ * BEGIN CANONICAL PROCESSED DOCSTRING
+ * Describes the possible states of an `ARTRealtimeChannel` object.
+ * END CANONICAL PROCESSED DOCSTRING
  */
 typedef NS_ENUM(NSUInteger, ARTRealtimeChannelState) {
+    /**
+     * BEGIN CANONICAL PROCESSED DOCSTRING
+     * The channel has been initialized but no attach has yet been attempted.
+     * END CANONICAL PROCESSED DOCSTRING
+     */
     ARTRealtimeChannelInitialized,
+    /**
+     * BEGIN CANONICAL PROCESSED DOCSTRING
+     * An attach has been initiated by sending a request to Ably. This is a transient state, followed either by a transition to `ARTRealtimeChannelAttached`, `ARTRealtimeChannelSuspended`, or `ARTRealtimeChannelFailed`.
+     * END CANONICAL PROCESSED DOCSTRING
+     */
     ARTRealtimeChannelAttaching,
+    /**
+     * BEGIN CANONICAL PROCESSED DOCSTRING
+     * The attach has succeeded. In the attached state a client may publish and subscribe to messages, or be present on the channel.
+     * END CANONICAL PROCESSED DOCSTRING
+     */
     ARTRealtimeChannelAttached,
+    /**
+     * BEGIN CANONICAL PROCESSED DOCSTRING
+     * A detach has been initiated on an `ARTRealtimeChannelAttached` channel by sending a request to Ably. This is a transient state, followed either by a transition to `ARTRealtimeChannelDetached` or `ARTRealtimeChannelFailed`.
```

Comment:
Can messages be published or be subscribed to at this state? It would be nice to clarify

Reply:
https://github.com/ably/ably-cocoa/pull/1491#discussion_r962945136